--- a/cmake/libprotoc.cmake
+++ b/cmake/libprotoc.cmake
@@ -209,10 +209,14 @@
   ${protobuf_source_dir}/src/google/protobuf/compiler/js/well_known_types/timestamp.js
 )
 add_executable(js_embed ${protobuf_source_dir}/src/google/protobuf/compiler/js/embed.cc)
+set(JS_EMBED_EXEC "js_embed")
+if(DEFINED ENV{EXTERNAL_JS_EMBED})
+  set(JS_EMBED_EXEC "$ENV{EXTERNAL_JS_EMBED}")
+endif()
 add_custom_command(
   OUTPUT ${protobuf_source_dir}/src/google/protobuf/compiler/js/well_known_types_embed.cc
   DEPENDS js_embed ${js_well_known_types_sources}
-  COMMAND js_embed ${js_well_known_types_sources} > ${protobuf_source_dir}/src/google/protobuf/compiler/js/well_known_types_embed.cc
+  COMMAND ${JS_EMBED_EXEC} ${js_well_known_types_sources} > ${protobuf_source_dir}/src/google/protobuf/compiler/js/well_known_types_embed.cc
 )

 add_library(libprotoc ${protobuf_SHARED_OR_STATIC}
